src/common/di.js

```diff
@@ -1,3 +1,124 @@
-import 'core-js/features/reflect/index.js';
+const emptyList = [];
```

Comment:
This empty list has no value and could be a source for trouble in the future. Why not just use [] where needed (I think there are 2 spots) that way we won't have any future trouble :).

Reply:
I put it there to avoid creating new arrays every time we resolve a dependency (it is only used when resolving it the first time to check for circular dependencies). If you think that it might cause trouble in the future then I'd rather put `null` there so we do not have that cost (it's in the performance critical path, called a lot of times during rendering).

Reply:
Done